LMChatCore_iOS/LMChatCore_iOS/Source/Module/Poll/CreatePoll/LMChatCreatePollDataModel.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  LMChatCreatePollDataModel.swift
+//  LikeMindsChatCore
+//
+//  Created by Pushpendra Singh on 17/07/24.
+//
+
+import Foundation
+
+public enum LMChatPollSelectState: Int, CustomStringConvertible, CaseIterable {
+    case exactly
+    case atMax
+    case atLeast
+    
+    public var description: String {
+        switch self {
+        case .exactly:
+            return "Exactly"
+        case .atMax:
+            return "At most"
+        case .atLeast:
+            return "At least"
+        }
+    }
+    
+    public var apiKey: String {
+        switch self {
+        case .exactly:
+            return "exactly"
+        case .atMax:
+            return "at_max"
+        case .atLeast:
+            return "at_least"
+        }
+    }
+    
+    public func checkValidity(with count: Int, allowedCount: Int) -> Bool {
+        switch self {
+        case .exactly:
+            return count == allowedCount
+        case .atMax:
+            return count <= allowedCount
+        case .atLeast:
+            return count >= allowedCount
+        }
+    }
+}
+
+extension LMChatPollSelectState {
+    init?(key: String) {
+        guard let type = LMChatPollSelectState.allCases.first(where: { $0.apiKey == key }) else { return nil }
+        self = type
+    }
+}
+
+public struct LMChatCreatePollDataModel {
+    public enum MetaOptions: Int, CustomStringConvertible, CaseIterable {
+        case isAnonymousPoll
+        case isInstantPoll
+        case allowAddOptions
+        
+        public var description: String {
```

Comment:
We can initialise enums with a value associated with them

e.g.

```
enum ProductCategory : String {
     case Washers = "washers", Dryers = "dryers", Toasters = "toasters"

     static let allValues = [Washers, Dryers, Toasters]
}
```

This might reduce our dependency on the description var that we defined explicitly

Reply:
Since, we are using Int enum, and we need to get text on the basis of particular case, that's why we are using description property. And if we assign direct string value to case it will not work for int raw value.